frb_codegen/src/generator/dart/mod.rs

```diff
@@ -394,14 +414,27 @@ fn generate_api_func(func: &IrFunc) -> GeneratedApiFunc {
         _ => format!(
             "{} => {}(FlutterRustBridgeTask(
             callFfi: (port_) => inner.{}({}),
-            parseSuccessData: _wire2api_{},
+            parseSuccessData: {},
             {}
         ));",
             partial,
             execute_func_name,
             func.wire_func_name(),
             wire_param_list.join(", "),
-            func.output.safe_ident(),
+            if (is_static_method(&func.name)
+                && static_method_return_struct_name(&func.name) == {
+                    if let IrType::StructRef(IrTypeStructRef { name, freezed: _ }) = &func.output {
+                        name.clone()
+                    } else {
+                        "".to_string()
+                    }
+                })
+                || struct_has_methods(ir_file, func.inputs.get(0).as_ref().map(|x| &x.ty))
+            {
+                format!("(d) => _wire2api_{}(this, d)", func.output.safe_ident())
+            } else {
+                format!("_wire2api_{}", func.output.safe_ident())
+            },
```

Comment:
nit: maybe extract to a function or at least not here inside an argument of `format!`